pkg/pxc/app/pmm.go

```diff
@@ -52,9 +53,125 @@ func PMMClient(spec *api.PMMSpec, secrets string, v120OrGreater bool) corev1.Con
 		container.Ports = ports
 	}
 
+	if v170OrGreater {
+		container.LivenessProbe = &corev1.Probe{
+			InitialDelaySeconds: 60,
+			TimeoutSeconds:      5,
+			PeriodSeconds:       10,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Port: intstr.FromInt(7777),
+					Path: "/local/Status",
+				},
+			},
+		}
+		container.Env = append(container.Env, pmmAgentEnvs(spec.ServerHost, spec.ServerUser, secrets)...)
+		container.Lifecycle = &corev1.Lifecycle{
+			PreStop: &corev1.Handler{
+				Exec: &corev1.ExecAction{
+					Command: []string{"bash", "-c", "pmm-admin inventory remove node --force $(pmm-admin status --json | python -c \"import sys, json; print(json.load(sys.stdin)['pmm_agent_status']['node_id'])\")"},
+				},
+			},
+		}
+	}
+
 	return container
 }
 
+func pmmAgentEnvs(pmmServerHost, pmmServerUser, secrets string) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+		{
+			Name: "POD_NAMESPASE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name:  "PMM_AGENT_SERVER_ADDRESS",
+			Value: pmmServerHost,
+		},
+		{
+			Name:  "PMM_AGENT_SERVER_USERNAME",
+			Value: pmmServerUser,
+		},
+		{
+			Name: "PMM_AGENT_SERVER_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: SecretKeySelector(secrets, "pmmserver"),
+			},
+		},
+		{
+			Name:  "PMM_AGENT_LISTEN_PORT",
+			Value: "7777",
+		},
+		{
+			Name:  "PMM_AGENT_PORTS_MIN",
+			Value: "30100",
+		},
+		{
+			Name:  "PMM_AGENT_PORTS_MAX",
+			Value: "30105",
+		},
+		{
+			Name:  "PMM_AGENT_CONFIG_FILE",
+			Value: "/usr/local/percona/pmm2/config/pmm-agent.yaml",
+		},
+		{
+			Name:  "PMM_AGENT_SERVER_INSECURE_TLS",
+			Value: "1",
+		},
+		{
+			Name:  "PMM_AGENT_LISTEN_ADDRESS",
+			Value: "0.0.0.0",
+		},
+		{
+			Name:  "PMM_AGENT_SETUP_NODE_NAME",
+			Value: "$(POD_NAMESPASE)-$(POD_NAME)",
+		},
+		{
+			Name:  "PMM_AGENT_SETUP_METRICS_MODE",
+			Value: "push",
+		},
+		{
+			Name:  "PMM_AGENT_SETUP",
+			Value: "1",
+		},
+		{
+			Name:  "PMM_AGENT_SETUP_FORCE",
+			Value: "1",
+		},
+		{
+			Name:  "PMM_AGENT_SETUP_NODE_TYPE",
+			Value: "container",
+		},
+	}
+}
+
+func PMMAgentScript(dbType string) []corev1.EnvVar {
+	pmmServerArgs := " $(PMM_ADMIN_CUSTOM_PARAMS) --skip-connection-check --metrics-mode=push "
+	pmmServerArgs += " --username=$(DB_USER) --password=$(DB_PASSWORD) --cluster=$(CLUSTER_NAME) "
+	pmmServerArgs += " --service-name=$(PMM_AGENT_SETUP_NODE_NAME) --host=$(POD_NAME) --port=$(DB_PORT) "
+	if dbType == "mysql" {
+		pmmServerArgs += "$(DB_ARGS)"
+	}
+	return []corev1.EnvVar{
+		{
+			Name:  "PMM_AGENT_PRERUN_SCRIPT",
+			Value: "pmm-admin status --wait=10s; pmm-admin add $(DB_TYPE)" + pmmServerArgs + "; pmm-admin annotate --service-name=$(PMM_AGENT_SETUP_NODE_NAME) restart",
```

Comment:
It can be concatenated with \n to produce valid multiline YAML field. It doesn't have to be one-liner.

And "restart" there is a custom message. I guess it can be better / more user-friendly.

Reply:
which text is better?

Reply:
@AlekSi The concatenation was added, if you have better 'text' for custom message please suggest.

Reply:
Guys. Users will see that text in the UI. That's not for me to provide texts for users. I think you have product people for that :)

Reply:
@spron-in please suggest what text we can use for 'annotation'. Now we use 'restart' to identify that container was restarted.

Reply:
Lets put "Service restarted" message.

Reply:
ok

Reply:
@spron-in  @AlekSi  I have updated it 025f55e